signalduino_protocols.hash - update

HomeAutoUser opened the pull request:
* **Please check if the PR fulfills these requirements**
<!-- Please do not remove these checkboxes. Check them if you have done the action behind this point -->
- [x] CHANGED has been updated (needed for doc)


<!--Please specify if this is a bugfix, feature or update of docs and remove the other options -->

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
revised ID 90 - added doc https://github.com/RFD-FHEM/RFFHEM/issues/449


* **What is the current behavior?** (You can also link to an open issue here)
no doc

sidey79 commented:
Was ist mit ID30 passiert?

HomeAutoUser commented:
Da ich derzeit an dem Modul SD_UT arbeite und dies auch versuche noch anzupassen bin ich auf die ID 30 gestoßen weil diese ebenso wie andere auch "doppelt" dispatcht werden. So habe ich mir die ID 30 und 83 angesehen.
```

	"30"	=>	## a unitec remote door reed switch
			# https://forum.fhem.de/index.php?topic=43346.0
			# MU;P0=-10026;P1=-924;P2=309;P3=-688;P4=-361;P5=637;D=123245453245324532453245320232454532453245324532453202324545324532453245324532023245453245324532453245320232454532453245324532453202324545324532453245324532023245453245324532453245320232454532453245324532453202324545324532453245324532023240;CP=2;O;
						# MU;P0=307;P1=-10027;P2=-691;P3=-365;P4=635;D=0102034342034203420342034201020343420342034203420342010203434203420342034203420102034342034203420342034201020343420342034203420342010203434203420342034203420102034342034203420342034201;CP=0;
		{
			name			=> 'diverse',
			comment			=> 'remote control unitec | door reed switch 47031',
			id			=> '30',
			knownFreqs		=> '',
			one			=> [-2,1],
			zero			=> [-1,2],
			start			=> [-30,1],				# Message is not provided as MS, worakround is start
			clockabs		=> 330,						# ca 300 us
			format			=> 'twostate',		# there is a pause puls between words
			preamble		=> 'P30#',				# prepend to converted message
			clientmodule		=> 'SD_UT',
			modulematch		=> '^P30#.{3}',
			length_min		=> '12',
			length_max		=> '12',				# message has only 10 bit but is paddet to 12
		},
```
```

	"83"	=>	## Remote control RH787T based on MOSDESIGN SEMICONDUCTOR CORP (CMOS ASIC encoder) M1EN compatible HT12E
			# for example Westinghouse Deckenventilator Delancey, 6 speed buttons, @zwiebelxxl
			# https://github.com/RFD-FHEM/RFFHEM/issues/250
			# Taste 1 MU;P0=388;P1=-112;P2=267;P3=-378;P5=585;P6=-693;P7=-11234;D=0123035353535356262623562626272353535353562626235626262723535353535626262356262627235353535356262623562626272353535353562626235626262723535353535626262356262627235353535356262623562626272353535353562626235626262723535353535626262356262627235353535356262;CP=2;R=43;O;
			# Taste 2 MU;P0=-176;P1=262;P2=-11240;P3=112;P5=-367;P6=591;P7=-695;D=0123215656565656717171567156712156565656567171715671567121565656565671717156715671215656565656717171567156712156565656567171715671567121565656565671717156715671215656565656717171567156712156565656567171715671567121565656565671717171717171215656565656717;CP=1;R=19;O;
			# Taste 3 MU;P0=564;P1=-392;P2=-713;P3=245;P4=-11247;D=0101010101023231023232323431010101010232310232323234310101010102323102323232343101010101023231023232323431010101010232310232323234310101010102323102323232343101010101023231023232323431010101010232310232323234310101010102323102323232343101010101023231023;CP=3;R=40;O;
		{
			name			=> 'RH787T',
			comment			=> 'remote control for example Westinghouse Delancey 7800140',
			id			=> '83',
			knownFreqs		=> '',
			one			=> [-2,1],
			zero			=> [-1,2],
			start			=> [-35,1],				# calculated 12126,31579 µS
			clockabs		=> 335,						# calculated ca 336,8421053 µS short - 673,6842105µS long
			format			=> 'twostate',		# there is a pause puls between words
			preamble		=> 'P83#',				# prepend to converted message
			clientmodule	=> 'SD_UT',
			modulematch		=> '^P83#.{3}',
			length_min		=> '12',
			length_max		=> '12',
		},
```

Da bei einem Dispatch der ID 30 ebenso die 83 kam und auch bei der 83 die 30, so habe ich mir die Def´s angesehen. Da diese "gleich" waren, so habe ich die 30 "aufgelöst" weil dort man nicht sendet aufgrund das es nur ein door / switch Kontakt zum receive ist.

Somit habe ich die 30er Dokumentation in die 83er ergänzt und würde nun die 30er wie im Forum für ein neues Def nutzen wolllen. Einwende?

sidey79 commented:
Einwände würde ich nicht sagen. Die ID 30 sollte man halt nicht gleich morgen wieder verwenden finde ich.
Da die Daten an das Modul SD_WS übergeben wurden, gehe ich davon aus, dass die angelegten Definitionen auch weiterhin wie bisher  mit der ID 83 funktionieren?

HomeAutoUser commented:
Du meinst das SD_UT Modul :)
Bei solchen Änderungen und vorbehaltenen „Reservierungen“ wäre es hilfreich das wir uns einen Zeitraum benennen weil wir sonst immer mehr Protokolle anlegen obwohl wir „Löcher“ bei den fortlaufenden IDs haben. Sowas möchte ich vermeiden weil nur wenige schauen welche IDs frei wären für was neues.

sidey79 commented:
Hmm, bis zu einem neuen Release vileicht?

Ralf9 commented:
Mir ist nicht klar warum für den Remote control RH787T damals überhaupt eine neue ID 83 nötig war obwohl die ID 30 auch gepasst hat. 
Ich habe mir https://github.com/RFD-FHEM/RFFHEM/issues/250 mal angeschaut, anfangs wurde noch die ID 30 verwendet, dann wurde für mich im nachhinein nicht mehr nachvollziehbar die ID 83 verwendet.
Wäre es nicht sinnvoll, wie ursprünglich angedacht für die RH787T die ID 30 zu nehmen und die ID 83 zu löschen?

sidey79 commented:
Wenn ich es richtig verstanden habe, dann wird über ID 83 bereits gesendet und bei ID30 wird nur empfangen. Daher war wohl die Idee, die ID30 zu entfernen, da hier weniger implementiert wurde und somit die Gefahr von Problemen geringer ist.

Ralf9 commented:
Die Probleme dürften aber beim löschen von ID 30 für den Anwender größer sein.
Wenn er die ID 30 in die whitelist eingetragen hat und dann nach einem update seine Reed Kontakte nicht mehr funktionieren, wie soll er dann draufkommen, daß er nun die ID 83 in die whitelist eintragen muss?
Nach Murphy werden solche updates häufig dann gemacht, wenn man eigentlich keine Zeit zur Fehlersuche hat.

Beim Senden über die ID 83 dürfte es nach dem entfernen der ID 83 keine Probleme geben, wenn der Anwender über das Modul sendet. Nach dem Update des SD_UT Modul, sended es über die ID 30.

elektron-bbs commented:
Die Ident 83 haben wir dazu genommen, weil zumindest die Startwerte ja nicht so 100%ig zur 30 passen. Die Timings dieses Protokolls stimmen nach Datenblatt des Schaltkreises.
Bei der 30 sind es ja wahrscheinlich nur "Schätzwerte". Ich habe deshalb eigentlich auch Bedenken, die 30 zu löschen, auch wenn da jetzt manche Nachrichten doppelt dispatcht werden.

HomeAutoUser commented:
Es wurde damals eine neue ID83 verwendet weil es beim senden andere Werte gab wie die bisher dort implementierte ID30.

Bis dato war die 30 nicht in ein Modul implementiert und ich machte mir die Arbeit Kontakt mit dem User aufzunehmen. Die Verarbeitung der Kontakte, egal ob über DoIF oder via Modul sind NICHT 100% fertig weil der User sich von uns „Abwand“ bzw. Nicht mehr benötigt. Ein solches Verhalten tritt auf, wenn wir wünsche oder Anregungen mit der Implementierung nur „halb“ umsetzen.

Das ein anderer User die30 auswertet, halte ich fast für gen null. Gern irre ich mich da auch.

Ich sehe uns hier nur weiterhin auffüllen von Protokollen welche ggf „sinnvoll“ verbraucht werden könnten. Eine Prüfung mit mehreren zutreffenden IDs in dem oder einem Modul ist nur über die Länge möglich. Das größte Ansinnen dahinter ist, das ich / wir nicht den überblick der Verarbeitung verlieren.

Id30 / 83 ... gern belassen wir es so.

HomeAutoUser commented:
Thematik whitelist... mit sowas verbauen wir uns auch die Möglichkeit, dem User sinnvoll mitzuteilen, das die bei ihm Black / whitelist Option nun auf eine neue Definition wirkt. Das sehe ich als größtes Problem wo ich uns nur sehe, Protokolle hinzuzufügen aber nicht den älteren „Müll“ neu verwenden. Ich betrachte es aus der useransicht ohne Nutzung der Funktionen Black / whitelist. Ich sehe die Funktionen sogar als „Gefahr“ weil man so nicht richtig mitbekommt was sich ändert und somit wir getrieben werden alles drin zu lassen ohne Ordnung zu schaffen ;)

HomeAutoUser commented:
Ich habe nun nach der Diskussion den alten Stand geschaffen und nur die ID 91 hinzugefügt.

Das wir nicht wieder in unterschiedlichen Meinungen enden, wie seht Ihr das bei der Definition?
Die ID 90 ist derzeit mumbi unsw. Da bei  ESTO Lighting GmbH das selbe herauskommen würde nur mit einem anderen CP aber das Verhältnis gleich, so könnte man es in eine stecken. Wiederrum spricht dafür, das beim senden dann ggf. es einen Unterschied geben muss aufgrund der Syncs. richtig?

```
	"90"	=>	## mumbi m-FS300 / manax MX-RCS250 (CP 260-275)
			# https://forum.fhem.de/index.php/topic,94327.15.html
			# MS;P0=-9964;P1=273;P4=-866;P5=792;P6=-343;D=10145614141414565656561414561456561414141456565656561456141414145614;CP=1;SP=0;R=35;O;m2;		//A	AN
			# MS;P0=300;P1=-330;P2=-10160;P3=804;P7=-840;D=02073107070707313131310707310731310707070731313107310731070707070707;CP=0;SP=2;R=23;O;m1;	//A	AUS
			# MS;P1=260;P2=-873;P3=788;P4=-351;P6=-10157;D=16123412121212343434341212341234341212121234341234341234121212341212;CP=1;SP=6;R=21;O;m2;	//B	AN
			# MS;P1=268;P3=793;P4=-337;P6=-871;P7=-10159;D=17163416161616343434341616341634341616161634341616341634161616343416;CP=1;SP=7;R=24;O;m2;	//B	AUS
			## ESTO Lighting GmbH | remote control KL-RF01 with 9 buttons (CP 375-395)
			# https://github.com/RFD-FHEM/RFFHEM/issues/449
			# MS;P1=376;P4=-1200;P5=1170;P6=-409;P7=-12224;D=17141414561456561456565656145656141414145614141414565656145656565614;CP=1;SP=7;R=231;e;m0; 
			# MS;P1=393;P2=-1174;P4=1180;P5=-401;P6=-12222;D=16121212451245451245454545124545124545451212121212121212454545454512;CP=1;SP=6;R=243;e;m0;
			# MS;P0=397;P1=-385;P2=-1178;P3=1191;P4=-12230;D=04020202310231310231313131023131023131020202020202020231313131313102;CP=0;SP=4;R=250;e;m0;
		{
			name         => 'ESTO Lighting GmbH | mumbi | MANAX',
			comment      => 'remote control mumbi FS300, MANAX MX-RCS250, ESTO KL-RF01',
			id           => '90',
			knownFreqs   => '433.92',
			one          => [3,-1],
			zero         => [1,-3],
			sync         => [1,-38],
			clockabs     => -1,						# -1=auto	
			format       => 'twostate',
			preamble     => 'u90#',			
			length_min   => '33',
			length_max   => '36',		
			#developId    => 'p',
		},
```

Demzufolge müsste diese in 2 Def´s richtig? Wenn ja, so ist das auch ein klassischer Fall wo dann sofort wieder 2 Def´s anschlagen.Diese zu verarbeiten ist dann schwierig wenn es keine Checksumme gibt zum prüfen.

elektron-bbs commented:
Das Senden wird ja wahrscheinlich mit "clockabs => -1," (auto) sowieso nicht funktionieren, oder irre ich da? Woher soll der Takt genommen werden, wenn nicht aus "clockabs"?

HomeAutoUser commented:
> 
> 
> Das Senden wird ja wahrscheinlich mit "clockabs => -1," (auto) sowieso nicht funktionieren, oder irre ich da? Woher soll der Takt genommen werden, wenn nicht aus "clockabs"?

Sollte es so sein, so wäre eine Definition für eine Remote mit -1 ja unbrauchbar wenn wir diese versuchen in ein Modul einzubauen. Dazu kann uns nur @sidey79 aufklären.

sidey79 commented:
> Thematik whitelist... mit sowas verbauen wir uns auch die Möglichkeit, dem User sinnvoll mitzuteilen, das die bei ihm Black / whitelist Option nun auf eine neue Definition wirkt. Das sehe ich als größtes Problem wo ich uns nur sehe, Protokolle hinzuzufügen aber nicht den älteren „Müll“ neu verwenden. Ich betrachte es aus der useransicht ohne Nutzung der Funktionen Black / whitelist. Ich sehe die Funktionen sogar als „Gefahr“ weil man so nicht richtig mitbekommt was sich ändert und somit wir getrieben werden alles drin zu lassen ohne Ordnung zu schaffen ;)

Tia, ich weiss selbst nicht so recht, wie man damit umgehen sollte.
Wir könnten uns auch die Frage stellen, welcher Vorteil bringt das wiederverwenden alter IDs?

sidey79 commented:
> > Thematik whitelist... mit sowas verbauen wir uns auch die Möglichkeit, dem User sinnvoll mitzuteilen, das die bei ihm Black / whitelist Option nun auf eine neue Definition wirkt. Das sehe ich als größtes Problem wo ich uns nur sehe, Protokolle hinzuzufügen aber nicht den älteren „Müll“ neu verwenden. Ich betrachte es aus der useransicht ohne Nutzung der Funktionen Black / whitelist. Ich sehe die Funktionen sogar als „Gefahr“ weil man so nicht richtig mitbekommt was sich ändert und somit wir getrieben werden alles drin zu lassen ohne Ordnung zu schaffen ;)
> 
> Tia, ich weiss selbst nicht so recht, wie man damit umgehen sollte.
> Wir könnten uns auch die Frage stellen, welcher Vorteil bringt das wiederverwenden alter IDs?



> > Das Senden wird ja wahrscheinlich mit "clockabs => -1," (auto) sowieso nicht funktionieren, oder irre ich da? Woher soll der Takt genommen werden, wenn nicht aus "clockabs"?
> 
> Sollte es so sein, so wäre eine Definition für eine Remote mit -1 ja unbrauchbar wenn wir diese versuchen in ein Modul einzubauen. Dazu kann uns nur @sidey79 aufklären.

Die Clock kann in sendMsg z.B. als C400 angegeben werden.

Die ID90 passt für MS Nachrichten und 91 für MU. Da kann man nicht viel zusammenfassen. Ich bin aber auch immer wieder im Dilemma wie man sehr ähnliche Definitionen findet (siehe z.B. Krinner Protokoll)

HomeAutoUser commented:
> 
> > > Das Senden wird ja wahrscheinlich mit "clockabs => -1," (auto) sowieso nicht funktionieren, oder irre ich da? Woher soll der Takt genommen werden, wenn nicht aus "clockabs"?
> > 
> > 
> > Sollte es so sein, so wäre eine Definition für eine Remote mit -1 ja unbrauchbar wenn wir diese versuchen in ein Modul einzubauen. Dazu kann uns nur @sidey79 aufklären.
> 
> Die Clock kann in sendMsg z.B. als C400 angegeben werden.
> 
> Die ID90 passt für MS Nachrichten und 91 für MU. Da kann man nicht viel zusammenfassen. Ich bin aber auch immer wieder im Dilemma wie man sehr ähnliche Definitionen findet (siehe z.B. Krinner Protokoll)


In dem Falle geht es um diese Varianten:
```
	"90"	=>	## socket remote -  mumbi m-fs300 / manax MX-RCS250
						# https://forum.fhem.de/index.php/topic,94327.15.html
						# MS;P0=-9964;P1=273;P4=-866;P5=792;P6=-343;D=10145614141414565656561414561456561414141456565656561456141414145614;CP=1;SP=0;R=35;O;m2;		//A AN
						# MS;P0=300;P1=-330;P2=-10160;P3=804;P7=-840;D=02073107070707313131310707310731310707070731313107310731070707070707;CP=0;SP=2;R=23;O;m1;	//A	AUS
						# MS;P1=260;P2=-873;P3=788;P4=-351;P6=-10157;D=16123412121212343434341212341234341212121234341234341234121212341212;CP=1;SP=6;R=21;O;m2;	//B AN
						# MS;P1=268;P3=793;P4=-337;P6=-871;P7=-10159;D=17163416161616343434341616341634341616161634341616341634161616343416;CP=1;SP=7;R=24;O;m2;	//B	AUS
						# need more Device Infos / User Message
		{
			name         => 'm-fs300',
			comment      => 'IT remote control mumbi FS300, MX-RCS250',
			id           => '90',
			knownFreqs   => '433.92',
			one          => [3,-1],
			zero         => [1,-3],
			sync         => [1,-38],
			clockabs     => -1,						# -1=auto
			format       => 'twostate',
			preamble     => 'u90#',
			length_min   => '33',
			length_max   => '36',
			developId    => 'p',
		},
```

UND zusätzlich https://github.com/RFD-FHEM/RFFHEM/issues/449

```
	"92"	=>	## ESTO Lighting GmbH | remote control KL-RF01 with 9 buttons (CP 375-395)
						# https://github.com/RFD-FHEM/RFFHEM/issues/449
						# MS;P1=376;P4=-1200;P5=1170;P6=-409;P7=-12224;D=17141414561456561456565656145656141414145614141414565656145656565614;CP=1;SP=7;R=231;e;m0; 
						# MS;P1=393;P2=-1174;P4=1180;P5=-401;P6=-12222;D=16121212451245451245454545124545124545451212121212121212454545454512;CP=1;SP=6;R=243;e;m0;
						# MS;P0=397;P1=-385;P2=-1178;P3=1191;P4=-12230;D=04020202310231310231313131023131023131020202020202020231313131313102;CP=0;SP=4;R=250;e;m0;
		{
			name         => 'ESTO Lighting GmbH',
			comment      => 'remote control ESTO KL-RF01',
			id           => '92',
			knownFreqs   => '433.92',
			one          => [3,-1],
			zero         => [1,-3],
			sync         => [1,-38],
			clockabs     => -1,						# -1=auto	
			format       => 'twostate',
			preamble     => 'u92#',			
			length_min   => '33',
			length_max   => '36',		
			#developId    => 'p',
		},
```

Man könnte die beiden Zusammenfassen so --->

```
	"90"	=>	## mumbi m-FS300 / manax MX-RCS250 (CP 260-275)
						# https://forum.fhem.de/index.php/topic,94327.15.html
						# MS;P0=-9964;P1=273;P4=-866;P5=792;P6=-343;D=10145614141414565656561414561456561414141456565656561456141414145614;CP=1;SP=0;R=35;O;m2;		//A	AN
						# MS;P0=300;P1=-330;P2=-10160;P3=804;P7=-840;D=02073107070707313131310707310731310707070731313107310731070707070707;CP=0;SP=2;R=23;O;m1;	//A	AUS
						# MS;P1=260;P2=-873;P3=788;P4=-351;P6=-10157;D=16123412121212343434341212341234341212121234341234341234121212341212;CP=1;SP=6;R=21;O;m2;	//B	AN
						# MS;P1=268;P3=793;P4=-337;P6=-871;P7=-10159;D=17163416161616343434341616341634341616161634341616341634161616343416;CP=1;SP=7;R=24;O;m2;	//B	AUS
						## ESTO Lighting GmbH | remote control KL-RF01 with 9 buttons (CP 375-395)
						# https://github.com/RFD-FHEM/RFFHEM/issues/449
						# MS;P1=376;P4=-1200;P5=1170;P6=-409;P7=-12224;D=17141414561456561456565656145656141414145614141414565656145656565614;CP=1;SP=7;R=231;e;m0; 
						# MS;P1=393;P2=-1174;P4=1180;P5=-401;P6=-12222;D=16121212451245451245454545124545124545451212121212121212454545454512;CP=1;SP=6;R=243;e;m0;
						# MS;P0=397;P1=-385;P2=-1178;P3=1191;P4=-12230;D=04020202310231310231313131023131023131020202020202020231313131313102;CP=0;SP=4;R=250;e;m0;
		{
			name         => 'ESTO Lighting GmbH | mumbi | MANAX',
			comment      => 'remote control mumbi FS300, MANAX MX-RCS250, ESTO KL-RF01',
			id           => '90',
			knownFreqs   => '433.92',
			one          => [3,-1],
			zero         => [1,-3],
			sync         => [1,-38],
			clockabs     => -1,						# -1=auto	
			format       => 'twostate',
			preamble     => 'u90#',			
			length_min   => '33',
			length_max   => '36',		
			#developId    => 'p',
		},
```

und mir der Erkenntnis mit der Angabe von CP beim senden manuell anzugeben, das man dann im Modul den Befehl anpassen muss.

ODER doch 2 Varianten aber diese werden dann doppelt durchlaufen weil diese beide zutreffen aber der CP ist anders. (in klammern habe ich diesen mal notiert).

Ich plädiere ja fast dür 1 Definition und dann den CP manuell angeben im Modul.

sidey79 commented:
> Ich plädiere ja fast dür 1 Definition und dann den CP manuell angeben im Modul.

So wie das für mich aussieht, kann eine Definition verwendet werden.
 
> EDIT: Wenn ich das richtig im Kopf habe, so ist es zwar möglich manuell den CP anzugeben aber wenn ich diesen bei einer Doppeltnutzung definiere, so wird zur Berechnung ja die Angabe von sync unsw. genutzt davon. Da ist bei einer doppelten Nutzung wo sich der Sync ähnelt von Tolleranzen, ja dennoch dann der wert Falsch weil er FEst vom eintragen genutzt wird.... ---> Einzelne Definitionen wären da besser wieder.

Das habe ich leider nicht verstanden.
Den Takt kannst Du als Wert beim Aufruf von sendMSG mitgeben.
Im Logischen Modul gibt es meines Erachtens nach folgende Optionen

1. Das Modul ermittelt die Clock für das Senden aus RAWMSG und speichert es sich ab
2. Das Modul hat ein Attribut, was beim Autocreate auch schon vorbelegt werden kann und der Anwender kann es ändern (siehe IT Modul)
3. Das Modul hat ein Attribut Modell und damit wird der passende Takt ausgewählt.

HomeAutoUser commented:
Wenn wir die beiden Devices in eine Definition bringen, so sind wir gezwungen beides manuell auszuwerten weil wir den Match auf ein Modul machen und nicht auf das Model, oder können wir es weitergeben?

Mein Ansinnen wäre, das ELO Device ggf ins SD_UT einzubinden. Das andere , da waren wir uns ja noch noch schlüssig. Gern füge ich es ebenso ins SD_UT hinzu weil so ist es auch nur eine Anpassung in de Modul ohne Erweiterung vom IT Modul.

HomeAutoUser commented:
Auf diesen Stand des PR würde ich gern aufbauen wollen.
- added ID 91 (new Device Atlantic Security)
- added doc ID 90 + Ergänzung anderes Device (ESTO Lighting GmbH | remote control KL-RF01 with 9 buttons (CP 375-395)

Somit können wir es einbinden in das Modul.

HomeAutoUser commented:
@sidey79 
ich habe mir das ganze mit den doppelten Decoded nochmal angesehen und es unterschiedlich durchlaufen lassen mit dem von den Usern erhaltenen Nachrichten. Besser werden wir kommen diese getrennt zu behandeln weil dort die Zuverlässigkeit größer ist. Wir werden auszuverlässigkeit nicht drum herum kommen, wieder 2 verschiedene Defs zu machen. Anderenfalls kann es passieren, das bei dem User wieder zu viele MSG nicht decodiert werden. Eine von beiden Nachrichten des verschiedenen Geräte wird immer unterdrückt weil diese nicht zu 100% in die Tolleranz passt.
